### ISSUE TITLE
Infinite db restoration

### ISSUE THREAD

**Issue by PhantomYdn** [orientechnologies/orientdb#9226]
### OS: win10

After crash on non-production (yet) environment, orientDB was trying to recover db. Latest from logs:

```
2020-04-22 20:52:54:530 INFO  Looking for last checkpoint... [OLocalPaginatedStorage]
2020-04-22 20:52:54:530 INFO  FULL checkpoint found. [OLocalPaginatedStorage]
2020-04-22 20:52:54:535 INFO  Data restore procedure from full checkpoint is started. Restore is performed from LSN OLogSequenceNumber{segment=5, position=18} [OLocalPaginatedStorage]
```

And after such logs - process just stuck - nothing happening.
Moreover: DB is small and was created from scratch by 3.1-SNAPSHOT version.

Any ideas what's going on?

**Comment by laa** [comment]
Hi @PhantomYdn this functionality is under development. It will be fixed in 3 weeks.

**Comment by PhantomYdn** [comment]
@laa , thank you for clarifications!  Is this the same situation with scheduling? As we figured out, we are experiencing crashing of DB due to OScheduler functionality... Attempt to schedule tasks leads to hanging up forever and only after terminating DB and starting over - we see this issue listed above.

**Comment by laa** [comment]
> @laa , thank you for clarifications! Is this the same situation with scheduling? As we figured out, we are experiencing crashing of DB due to OScheduler functionality... Attempt to schedule tasks leads to hanging up forever and only after terminating DB and starting over - we see this issue listed above.

No those are different things.  If you don't mind please open a new issue.

**Comment by PhantomYdn** [comment]
@laa , sure. Separate one has been open: #9235 Apparantly it's even not a Scheduler, but OrientDB ThreadPool

**Comment by tglman** [comment]
Hi,

The 3.1.x has been released that should include fixes for this specific cases, could you double check this?

Regards

**Comment by laa** [comment]
Hi @PhantomYdn looking forward to your response.

**Comment by PhantomYdn** [comment]
@laa , @tglman , just checked on 3.1.0 - stuck on exactly the same point during restoration.
![image](https://user-images.githubusercontent.com/1199285/83703983-c9bd7b00-a5c5-11ea-8d1e-40e4da6ca22f.png)

As I wrote previously: result of the broken db is issue #9235 - I also checked it on 3.1.0 - exactly the same behavior.

**Comment by madmac2501** [comment]
I also reproduced this issue in 3.1.0

**Comment by laa** [comment]
Hi @PhantomYdn, unfortunately, I can not reproduce it. Actually I have been running tests for a week to reproduce it.  Could you send me a thread dump?

**Comment by madmac2501** [comment]
Hi @laa I have a thread dump in the ticket that I opened about this issue https://github.com/orientechnologies/orientdb/issues/9283

Please if you have time take a look at my other tickets, we cannot use 3.1.0 because of these defects.

Thanks!

**Comment by laa** [comment]
Hi @madmac2501 , that is really helpful, but could you create not single but 2, 3 thread dumps so I will check what is going on. Could also send me a list of files inside of a database. Maybe you have two different issues actually.

**Comment by madmac2501** [comment]
Hi,

I rolled back to 3.0.31. I'll try to setup again the environment and try to reproduce it. Please give me some time.

Please send me your email to share this new information.

Thanks!

**Comment by laa** [comment]
Thank you. You can use a.lomakin@sap.com.

**Comment by laa** [comment]
@PhantomYdn can not answer this question unfortunately not my module.

**Comment by laa** [comment]
Guys, there is a small delay in release, we are fixing the issue which we want to be in next hotfix.

**Comment by laa** [comment]
@PhantomYdn , @madmac2501 we released new version, could you check it?

**Comment by madmac2501** [comment]
I tested 3.1.1 and this defect got fixed for me, but while testing I detected another possible defects. I opened issues in github for them.

Thanks a lot for the great work!

**Comment by laa** [comment]
Thank you @madmac2501 , @PhantomYdn could you confirm that too?

**Comment by PhantomYdn** [comment]
I also tested on 3.1.1 - the original issue is not reproducible for me. Seems fixed. Thank you, Andrey!
